tempesta_fw/http.h

```diff
@@ -435,6 +486,37 @@ struct tfw_http_resp_t {
 #define TFW_HTTP_RESP_STR_START(r)	__MSG_STR_START(r)
 #define TFW_HTTP_RESP_STR_END(r)	((&(r)->body) + 1)
 
+/**
+ * Streamed part of response or request.
+ *
+ * @stream_lock		- lock between parse/stream operations;
+ * @rcv_conn		- connection receiving the stream;
+ * @off			- offset from the buffer part end;
+ * @proc_off		- offset of unprocessed data in @body
+ *
+ * It's not always possible to stream the message as soon as it arrives,
+ * receiver may not be ready for this particular message, e.g. if there is one
+ * more streamed and not fully received message in its queue. So the streamed
+ * message is actually get buffered until the receiver is ready. There are also
+ * some cases when a more of the message must be buffered before is can be
+ * processed, e.g. trailer header processing.
```

Comment:
If there are 2 streamed messages, then there are should be 2 incomplete `TfwHttpMsg` descriptors - why not to use them? We even can use parts of `TfwMsg->skb_head` list, possibly with some skb pointers in the list, to handle a message headers and only current stream buffer...